File: spring-boot-blog-api/src/main/java/com/blog/services/PostService.java
package com.blog.services;
import java.util.List;
import com.blog.payloads.PostDto;
import com.blog.payloads.PostResponse;

public interface PostService {
	public PostDto createPost(PostDto postDto, Integer userId, Integer categoryId);
	public PostDto updatePost(PostDto postDto, Integer postId);
	public void deletePost(Integer postId);
	public PostDto getPost(Integer postId);
	public PostResponse getPosts(Integer page, Integer size, String sortBy, String order);
	public List<PostDto> getPostsByCategory(Integer categoryId);
	public List<PostDto> getPostsByUser(Integer userId);
	public List<PostDto> searchPosts(String keyword);
}

File: blogging-app/src/services/post-service.js
import { axiosObj, privateAxiosObj } from "./Helper"

export const createPost = (post) => {
  return privateAxiosObj.post(`/user/${post.userId}/category/${post.categoryId}/posts`, post).then(response => response.data);
}

export const getPosts = (page=0, size=5, sortBy='postId', order='asc') => {
  return axiosObj.get(`posts?page=${page}&size=${size}&sortBy=${sortBy}&order=${order}`).then(response => response.data);
}

export const getPost = (postId) => {
  return axiosObj.get(`posts/${postId}`).then(response => response.data);
}

export const createComment = (comment, postId, userId) => {
  return privateAxiosObj.post(`/post/${postId}/user/${userId}/comments`, { content: comment}).then(response => response.data);
}

export const uploadPostImage = (image, postId) => {
  let formData = new FormData();
  formData.append("image", image);
  return privateAxiosObj.post(`/image/upload/${postId}`, formData, { headers: { 'Content-Type': 'multipart/form-data' }}).then(response => response.data);
}

export const getPostsByCategory = (categoryId) => {
  return axiosObj.get(`/category/${categoryId}/posts`).then(response => response.data);
}

export const getPostsByUser = (userId) => {
  return axiosObj.get(`/user/${userId}/posts`).then(response => response.data);
}

export const deletePostById = (postId) => {
  return privateAxiosObj.delete(`/posts/${postId}`).then(response => response.data);
}

export const updatePost = (post) => {
  return privateAxiosObj.put(`/posts/${post.postId}`, post).then(response => response.data);
}